lib/model/app_model/bkpr_listincome.dart

```diff
@@ -0,0 +1,121 @@
+
+        /// FIXME : Can we do tempbalance = (2 * tempbalance - debit + credit) / 2
+        tempBalance += temp.creditMsat;
+        tempBalance -= temp.debitMsat;
+        outputList.add(temp);
+      }
+    }
+    outputList.sort((a, b) => b.timeStamp.compareTo(a.timeStamp));
+    return AppListIncome(balance: tempBalance, incomes: outputList);
+  }
+}
+
+class ListIncome {
+  final int creditMsat;
+
+  final int debitMsat;
+
+  final String identifier;
+
+  final String? description;
+
+  final int timeStamp;
+
+  final String paymentType;
+
+  final String? txId;
+
+  final String? outpoint;
+
+  ListIncome({
+    required this.creditMsat,
+    required this.debitMsat,
+    required this.identifier,
+    this.description,
+    required this.timeStamp,
+    required this.paymentType,
+    this.txId,
+    this.outpoint,
+  });
+
+  factory ListIncome.fromJSON(Map<String, dynamic> json,
+      {bool snackCase = true}) {
+    var creditMsat = json.withKey("credit_msat", snackCase: snackCase);
+    var debitMsat = json.withKey("debit_msat", snackCase: snackCase);
+    var tag = json.withKey("tag", snackCase: snackCase);
+    var timestamp = json.withKey("timestamp", snackCase: snackCase);
+    if (tag == "invoice") {
+      var description = json.withKey("description", snackCase: snackCase);
+      var identifier = "invoice";
+      var paymentForm = "off-chain";
+
+      /// This is a payment Id (preimage)
+      var txId = json.withKey("payment_id", snackCase: snackCase) ??
+          "Payment Id not found";
+
+      return ListIncome(
+          creditMsat: creditMsat,
+          debitMsat: debitMsat,
+          description: description,
+          identifier: identifier,
+          timeStamp: timestamp,
+          paymentType: paymentForm,
+          txId: txId);
+    } else if (tag == "deposit") {
+      var identifier = "wallet";
+      var paymentForm = "On-chain";
+      var outpoint = json.withKey("outpoint", snackCase: snackCase);
+
+      return ListIncome(
+          creditMsat: creditMsat,
+          debitMsat: debitMsat,
+          timeStamp: timestamp,
+          identifier: identifier,
+          paymentType: paymentForm,
+          outpoint: outpoint);
+    } else if (tag == "withdrawal") {
+      var identifier = "withdraw";
+      var paymentForm = "On-chain";
+      var outpoint = json.withKey("outpoint", snackCase: snackCase);
+
+      return ListIncome(
+          creditMsat: creditMsat,
+          debitMsat: debitMsat,
+          timeStamp: timestamp,
+          identifier: identifier,
+          paymentType: paymentForm,
+          outpoint: outpoint);
+    }
+    var identifier = "fee";
+    var paymentForm = "On-chain-fee";
+    var txId = json.withKey("txid", snackCase: snackCase);
+
+    return ListIncome(
+      creditMsat: creditMsat,
+      debitMsat: debitMsat,
+      timeStamp: timestamp,
+      identifier: identifier,
+      paymentType: paymentForm,
+      txId: txId,
+    );
+  }
```

Comment:
You are missing an event regarding the fee that it is invoice fee

from my testnet node I found

```
➜  ~ clightning --testnet bkpr-listincome | grep fee
         "tag": "invoice_fee",
         "tag": "onchain_fee",
         "tag": "onchain_fee",
         "tag": "onchain_fee",

```

Reply:
Actually I did not have this in my node. So I missed it

Reply:
yeah no problem, we should support it